docs/standard/base-types/regular-expression-source-generators.md

```diff
@@ -0,0 +1,400 @@
+
+```csharp
+private static readonly Regex s_abcOrDefGeneratedRegex =
+    new(pattern: "abc|def",
+        options: RegexOptions.Compiled | RegexOptions.IgnoreCase);
+
+private static void EvaluateText(string text)
+{
+    if (s_abcOrDefGeneratedRegex.IsMatch(text))
+    {
+        // Take action with matching text
+    }
+}
+```
+
+You can now rewrite the previous code as follows:
+
+```csharp
+private static readonly Regex s_abcOrDefGeneratedRegex = AbcOrDefGeneratedRegex();
+
+[GeneratedRegex("abc|def", RegexOptions.IgnoreCase | RegexOptions.Compiled, "en-US")]
+private static partial Regex AbcOrDefGeneratedRegex();
+
+private static void EvaluateText(string text)
+{
+    if (s_abcOrDefGeneratedRegex.IsMatch(text))
+    {
+        // Take action with matching text
+    }
+}
+```
+
+The generated implementation of `AbcOrDefGeneratedRegex()` similarly caches a singleton `Regex` instance, so no additional caching is needed to consume code.
+
+:::image type="content" source="media/regular-expression-source-generators/cached-instance.png" alt-text="Cached regex static field":::
+
+But as can be seen, it's not just doing `new Regex(...)`. Rather, the source generator is emitting as C# code a custom `Regex`-derived implementation with logic similar to what `RegexOptions.Compiled` emits in IL. You get all the throughput performance benefits of `RegexOptions.Compiled` (more, in fact) and the start-up benefits of `Regex.CompileToAssembly`, but without the complexity of `CompileToAssembly`. The source that's emitted is part of your project, which means it's also easily viewable and debuggable.
+
+:::image type="content" source="media/regular-expression-source-generators/debuggable-source.png" lightbox="media/regular-expression-source-generators/debuggable-source.png" alt-text="Debugging through source-generated Regex code":::
+
+> [!TIP]
+> In Visual Studio, select the project node in **Solution Explorer**, then expand **Dependencies** > **Analyzers** > **System.Text.RegularExpressions.Generator** > **System.Text.RegularExpressions.Generator.RegexGenerator** > _RegexGenerator.g.cs_ to see the generated C# code from this regex generator.
+
+You can set breakpoints in it, you can step through it, and you can use it as a learning tool to understand exactly how the regex engine is processing your pattern with your input. The generator even generates [triple-slash (XML) comments](../../csharp/language-reference/xmldoc/index.md) to help make the expression understandable at a glance and where it's used.
+
+:::image type="content" source="media/regular-expression-source-generators/xml-comments.png" lightbox="media/regular-expression-source-generators/xml-comments.png" alt-text="Generated XML comments describing regex":::
+
+## Inside the source-generated files
```

Comment:
Much of this section is about implementation detail.  Is it all relevant to the docs?